lib/restforce/concerns/api.rb

```diff
@@ -360,9 +360,13 @@ def upsert(*args)
       # Raises an exception if an error is returned from Salesforce, including the 300
       # error returned if the external ID provided matches multiple records (in which
       # case the conflicting IDs can be found by looking at the response on the error)
-      def upsert!(sobject, field, attrs)
+      def upsert!(sobject, field, attrs_arg)
+        attrs = attrs_arg.dup
```

Comment:
Feels a bit ugly having the argument named `attrs_arg` - could we just do the following? Should work fine I think? Just tried and it worked fine for me playing in `irb`.

```ruby
def upsert!(sobject, field, attrs)
  attrs = attrs.dup
  # ...
end